iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -316,56 +319,85 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
 // into
 //
 // %z = "BinaryElementwiseOpT"(%x, %y) : (%[[SHAPE_BEFORE_BCAST]], %[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_BEFORE_BCAST]]
-// %result = "mhlo.broadcast_in_dim"(%z) {} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
+// %result = "mhlo.broadcast_in_dim"(%z) {broadcast_dimensions = %[[BCAST_DIMS]]} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
 //
 // clang-format on
-template <typename BinaryElementwiseOpT>
-class ReorderBroadcastInDimOpAndBinaryElementwiseOp
-    : public OpRewritePattern<BinaryElementwiseOpT> {
+template <typename ElementwiseOpT>
+class ReorderBroadcastInDimOpAndElementwiseOp
+    : public OpRewritePattern<ElementwiseOpT> {
  public:
-  using OpRewritePattern<BinaryElementwiseOpT>::OpRewritePattern;
+  using OpRewritePattern<ElementwiseOpT>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(BinaryElementwiseOpT op,
+  LogicalResult matchAndRewrite(ElementwiseOpT op,
                                 PatternRewriter &rewriter) const override {
-    auto lhsType = op.lhs().getType().template dyn_cast<ShapedType>();
-    auto rhsType = op.rhs().getType().template dyn_cast<ShapedType>();
-    if (!lhsType || !rhsType || !lhsType.hasStaticShape() ||
-        lhsType != rhsType) {
-      return failure();
+    Operation *operation = op.getOperation();
+    // Sanity check: all elementwise ops shoule have 1 or more operand, only 1
+    // result, and no attribute.
```

Comment:
I think this comment can be removed because just describing the statements below, which isn't really helpful.

https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts